librz/core/linux_heap_glibc.c

```diff
@@ -1249,6 +1252,12 @@ void GH(print_malloc_states)(RzCore *core, GHT m_arena, MallocState *main_arena,
 		while (GH(is_arena)(core, m_arena, ta->GH(next)) && ta->GH(next) != m_arena) {
 			ut64 ta_addr = ta->GH(next);
 			if (!GH(rz_heap_update_main_arena)(core, ta->GH(next), ta)) {
+				if (json) {
+					pj_end(pj);
+					pj_end(pj);
+					rz_cons_println(pj_string(pj));
+					pj_free(pj);
+				}
 				free(ta);
 				return;
```

Comment:
maybe `goto end` and put `end:` near the end of the function where all these things are already done?